env0/resource_oci_credentials.go

```diff
@@ -0,0 +1,116 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceOciCredentials() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceOciCredentialsCreate,
+		UpdateContext: resourceOciCredentialsUpdate,
+		ReadContext:   resourceCredentialsRead(OCI_TYPE),
+		DeleteContext: resourceCredentialsDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceCredentialsImport(OCI_TYPE)},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name for the oci credentials",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"tenancy_ocid": {
+				Type:        schema.TypeString,
+				Description: "OCI tenancy OCID",
+				Required:    true,
+			},
+			"user_ocid": {
+				Type:        schema.TypeString,
+				Description: "OCI user OCID",
+				Required:    true,
+			},
+			"fingerprint": {
+				Type:        schema.TypeString,
+				Description: "OCI API key fingerprint",
+				Required:    true,
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Description: "OCI API private key",
+				Required:    true,
+				Sensitive:   true,
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Description: "OCI region",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "the env0 project id to associate the credentials with",
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
Assigning existing credentials to a new project should not re-create the credentials

Reply:
sure... I can remove it - but based on what I've been told, it should re-create when changing project id.
Let me know if I should proceed with removing this line.

Reply:
I see that all other creds are also following this principle

So in the case - let's continue and wait for feedback if it becomes an issue.